packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1035,7 +1031,8 @@ export class TransactionController extends BaseController<
     if (origin) {
       await validateTransactionOrigin(
         await this.getPermittedAccounts(origin),
-        this.getSelectedAddress(),
+        this.messagingSystem.call('AccountsController:getSelectedAccount')
```

Comment:
I feel like having a private method `#getSelectedAccount` would make it a bit clearer, WDYT?